zerver/lib/retention.py

```diff
@@ -303,15 +303,7 @@ def archive_personal_and_huddle_messages(realm: Realm, chunk_size: int=MESSAGE_B
     message_count = move_expired_personal_and_huddle_messages_to_archive(realm, chunk_size)
     logger.info("Done. Archived %s messages", message_count)
 
-def archive_stream_messages(realm: Realm, chunk_size: int=MESSAGE_BATCH_SIZE) -> None:
-    # We don't archive, if the stream has message_retention_days set to -1,
-    # or if neither the stream nor the realm have a retention policy.
-    query = Stream.objects.select_related("recipient").filter(
```

Comment:
It looks like we lost the `select_related("recipient")` -- I would expect that to be import?

Reply:
Oh yeah, that's important not to fetch `stream.recipient` from the database in a separate query per stream :neutral_face:  Thanks!